### DISCUSSION TITLE
Update answeroptions

### DISCUSSION THREAD

**Discussion by gorestea** [edgarrmondragon/citric#1185] [Q&A] [answered]
Hello! How to update answeroptions? I want to change the answer option from the drop-down list from test 1 to test 3. I called get_question_properties, there is an answeroptions parameter there. I tried to change it via set_question_properties, but I get the error No valid Data. May you help me?

**Comment by edgarrmondragon** [comment]
Hi @gorestea, I think this is essentially the same as https://github.com/edgarrmondragon/citric/discussions/1131. Can you confirm?

**Reply by gorestea** [reply]
It looks like. However, I would like to know if the limesurvey api provides such an opportunity? It would be possible to try through a http request with parameters. Have you tried it?

**Reply by edgarrmondragon** [reply]
The RemoteControl API certainly doesn't support this. It might be worth exploring the REST API for this. I don't have time to explore that option at the moment, but sending a PATCH request to with `questionAnswer` as `entity might do the trick.

https://github.com/edgarrmondragon/citric/issues/1186 might be a solution.

- OpenAPI spec: https://raw.githubusercontent.com/LimeSurvey/LimeSurvey/master/docs/open-api/v1.json
- https://citric.readthedocs.io/en/1.0.1/_api/citric/_rest/client/index.html

**Reply by gorestea** [reply]
How to work with REST API? How to connect it? There is nothing at mydomain.com/rest/v1. And in the settings in LM admin there are only RemoteControlAPI and RPC-interface

**Comment by edgarrmondragon** [comment]
Hello there! 👋

I just v2.0.0 of the library, which includes a new method to update answer options in a question.

Here's a quick example:

```python
from citric.rest import RESTClient

client = RESTClient(
    "https://example.com/rest/v1",
    "your_username",
    "your_password",
)

survey_id = 142423

# Get survey details
survey = client.get_survey(survey_id)
print(survey)

# Inspect question 247 to find answer IDs
for question in survey["questionGroups"][0]["questions"]:
    if answers := question.get("answers"):
        print("Answers for question", question["qid"])
        for answer in answers:
            print(f"  Answer ID: {answer['aid']}, Code: {answer['code']}")

# Define patch operations
operations = [
    {
        "entity": "answer",
        "op": "update",
        # Question ID
        "id": 247,
        "props": [
            {
                # Answer ID
                "aid": 31,
                "code": "AO01",
                "sortOrder": 1,
                "assessmentValue": 0,
                "scaleId": 1,
                "l10ns": {
                    "de": {"answer": "ANTW1 scale 1", "language": "de"},
                    "en": {"answer": "ANS1 scale 1", "language": "en"},
                },
            },
            {
                # Answer ID
                "aid": 32,
                "code": "AO02",
                "sortOrder": 1,
                "assessmentValue": 0,
                "scaleId": 1,
                "l10ns": {
                    "de": {"answer": "ANTW2 scale 1", "language": "de"},
                    "en": {"answer": "ANS2 scale 1", "language": "en"},
                },
            },
            {
                # Answer ID
                "aid": 33,
                "code": "AO03",
                "sortOrder": 1,
                "assessmentValue": 0,
                "scaleId": 1,
                "l10ns": {
                    "de": {"answer": "ANTW3 scale 1", "language": "de"},
                    "en": {"answer": "ANS3 scale 1", "language": "en"},
                },
            },
        ],
    }
]


result = client.patch_survey(survey_id, operations)
print(f"Operations applied: {result['operationsApplied']}")  # Operations applied: 1
```

It's also [available in the docs](https://citric.readthedocs.io/en/latest/how-to.html#update-question-answers-using-the-rest-api).

I would love to hear your feedback!

N.B. I'd like to add a more high-level API for this, but the LimeSurvey REST API seems to still be in-flux, so I'm waiting for it to stabilize before doing so.

**Comment by edgarrmondragon** [comment, marked as answer]
https://github.com/edgarrmondragon/citric/pull/1676 will let you do this

```python
dropdown = Question(
    title="Q03",
    type="L",
    l10ns={
        "en": QuestionL10n(question="Select your preferred framework"),
        "de": QuestionL10n(question="Wählen Sie Ihr bevorzugtes Framework"),
    },
    answer_options=[
        AnswerOption(
            code="DJ",
            l10ns={"en": "Django", "de": "Django"},
            sort_order=1,
        ),
        AnswerOption(
            code="FA",
            l10ns={"en": "FastAPI", "de": "FastAPI"},
            sort_order=2,
        ),
        AnswerOption(
            code="FL",
            l10ns={"en": "Flask", "de": "Flask"},
            sort_order=3,
        ),
    ],
)

client.import_question(dropdown.to_lsq(), survey_id, group_id)
```

This will be available in version 2.1.0.
